### PR TITLE
[CALCITE-4040] An aggregate function that does not support roll up throws an exception when it is rolled up

### PR DESCRIPTION
**Pull request by xy2953396112**
https://issues.apache.org/jira/browse/CALCITE-4040

### CODE REVIEW

File: core/src/test/java/org/apache/calcite/test/MaterializedViewSubstitutionVisitorTest.java [hunk 1]
```diff
@@ -410,6 +410,18 @@
     sql(mv, query).noMat();
   }
```
Comment (jinxing64) [inline core/src/test/java/org/apache/calcite/test/MaterializedViewSubstitutionVisitorTest.java:412]: How about some doc here to explain why matching should fail.

File: core/src/main/java/org/apache/calcite/plan/SubstitutionVisitor.java [hunk 2]
```diff
@@ -1899,8 +1899,12 @@ public static MutableRel unifyAggregates(MutableAggregate query,
         if (i < 0) {
           return null;
         }
+        SqlAggFunction aggFunction = getRollup(aggregateCall.getAggregation());
```
Comment (jinxing64) [inline core/src/main/java/org/apache/calcite/plan/SubstitutionVisitor.java:1902]: How about some doc to explain why matching should fail when the function doesn't support roll-up.

Comment (jinxing64) [reply core/src/main/java/org/apache/calcite/plan/SubstitutionVisitor.java:1902]: Mark it as `final`

File: core/src/test/java/org/apache/calcite/test/MaterializedViewSubstitutionVisitorTest.java [hunk 3]
```diff
@@ -410,6 +410,21 @@
...
 
+  /**
+   * stddev_pop aggregate function does not support roll up.
+   */
+  @Test public void testSqlAggFunctionRollup() {
```
Comment (jinxing64) [inline core/src/test/java/org/apache/calcite/test/MaterializedViewSubstitutionVisitorTest.java:416]: In current `MaterializedViewSubstitutionVisitorTest.java`, there's already an `testAggregateRollUp`. How about rename this as `testSqlAggFunctionRollup2` and put it after `testAggregateRollUp` ?

**Comment by xy2953396112** [comment]
@jinxing64 Thanks for review.

File: core/src/test/java/org/apache/calcite/test/MaterializedViewSubstitutionVisitorTest.java [hunk 4]
```diff
@@ -509,6 +509,21 @@
         .ok();
   }
 
+  /**
+   * stddev_pop aggregate function does not support roll up.
+   */
+  @Test public void testAggregateRollUp2() {
```
Comment (jinxing64) [inline core/src/test/java/org/apache/calcite/test/MaterializedViewSubstitutionVisitorTest.java:515]: `public` is not needed. Please remove.

**Pull request merged**